engine/src/test/java/org/camunda/bpm/engine/test/api/mgmt/ManagementServiceGetTelemetryDataTest.java

```diff
@@ -0,0 +1,396 @@
+    // given
+    MetricsRegistry metricsRegistry = configuration.getMetricsRegistry();
+    // create command data
+    metricsRegistry.markTelemetryOccurrence(ACTIVTY_INSTANCE_START, 5);
+
+    // when
+    TelemetryData firstTelemetryData = managementService.getTelemetryData();
+    TelemetryData secondTelemetryData = managementService.getTelemetryData();
+
+    assertThat(firstTelemetryData.getProduct().getInternals().getMetrics().get(ACTIVTY_INSTANCE_START).getCount()).isEqualTo(5);
+    assertThat(secondTelemetryData.getProduct().getInternals().getMetrics().get(ACTIVTY_INSTANCE_START).getCount()).isEqualTo(5);
+  }
+
+  @Test
+  @Deployment(resources = "org/camunda/bpm/engine/test/api/oneTaskProcess.bpmn20.xml")
+  public void shouldCollectMetrics_TelemetryEnabled() {
+    // given
+    managementService.toggleTelemetry(true);
+
+    // when
+    TelemetryData telemetryDataBeforePiStart = managementService.getTelemetryData();
+    engineRule.getRuntimeService().startProcessInstanceByKey("oneTaskProcess");
+    TelemetryData telemetryDataAfterPiStart = managementService.getTelemetryData();
+
+    // then
+    assertThat(telemetryDataBeforePiStart.getProduct().getInternals().getMetrics().get(ROOT_PROCESS_INSTANCE_START).getCount()).isEqualTo(0);
+    assertThat(telemetryDataAfterPiStart.getProduct().getInternals().getMetrics().get(ROOT_PROCESS_INSTANCE_START).getCount()).isEqualTo(1);
+  }
+
+  @Test
+  @Deployment(resources = "org/camunda/bpm/engine/test/api/oneTaskProcess.bpmn20.xml")
+  public void shouldCollectMetrics_TelemetryDisabled() {
+    // given
+    managementService.toggleTelemetry(false);
+
+    // when
+    TelemetryData telemetryDataBeforePiStart = managementService.getTelemetryData();
+    engineRule.getRuntimeService().startProcessInstanceByKey("oneTaskProcess");
+    TelemetryData telemetryDataAfterPiStart = managementService.getTelemetryData();
+
+    // then
+    Metric rootPiStartMetricBeforePiStart = telemetryDataBeforePiStart.getProduct().getInternals().getMetrics().get(ROOT_PROCESS_INSTANCE_START);
+    Metric rootPiStartMetricAfterPiStart = telemetryDataAfterPiStart.getProduct().getInternals().getMetrics().get(ROOT_PROCESS_INSTANCE_START);
+
+    assertThat(rootPiStartMetricBeforePiStart.getCount()).isEqualTo(0);
+    assertThat(rootPiStartMetricAfterPiStart.getCount()).isEqualTo(1);
+  }
+
+  @Test
+  @Deployment(resources = "org/camunda/bpm/engine/test/api/oneTaskProcess.bpmn20.xml")
+  public void shouldCollectCommands_TelemetryEnabled() {
+    // given
+    managementService.toggleTelemetry(true);
+
+    // when
+    TelemetryData telemetryDataBeforePiStart = managementService.getTelemetryData();
+    // trigger Command invocation
+    managementService.isTelemetryEnabled();
+    TelemetryData telemetryDataAfterPiStart = managementService.getTelemetryData();
+
+    // then
+    assertThat(telemetryDataBeforePiStart.getProduct().getInternals().getCommands().get(IS_TELEMETRY_ENABLED_CMD_NAME)).isNull();
```

Comment:
On a similar note, this line is not needed for this test: There should be another test that ensures that commands that were not executed are not counted. Your test can rely on this. If you want to verify the test setup, do it in the `given` block. That said, our codebase mostly does not verification of test setups (and on a related note, JUnit has the [`assume*` methods](https://junit.org/junit4/javadoc/4.13/org/junit/Assume.html) for this purpose, but we also don't use them either).

Same as the comment above, this is not critical for finishing this feature.